.babelrc.js

```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  presets: [
+    ['env', {
+      modules: process.env.BABEL_ENV === 'es'
```

Comment:
Can this not be accomplished with the env parameter used below? Something like

```
env: {
  es: {
    presets: [
      ['env', { modules: false }]
    ]
  },
  es: {
    presets: [
      ['env', { modules: 'commonjs' }]
    ]
  }
}
```

Didn't test this yet, but that way we don't have to move to a `.js` file just to access process env right?

Reply:
Using `env` for that changes the order plugins and presets are applied. Especially the class-properties transform causes problems, because the `env` preset throws an error if the class-properties transform was not applied yet.

The entire config would have to be duplicated for each env to maintain the correct order:

```json
{
  "env": {
    "es": {
      "presets": [
        ["env", { "modules": false }],
        "react"
      ],
      "plugins": [
        "transform-object-rest-spread",
        "transform-class-properties",
        "transform-export-extensions",
        "transform-object-assign"
      ]
    },
    "commonjs": {
      "presets": [
        ["env", { "modules": "commonjs" }],
        "react"
      ],
      "plugins": [
        "transform-object-rest-spread",
        "transform-class-properties",
        "transform-export-extensions",
        "transform-object-assign"
      ]
    },
    "test": {
      "presets": [
        ["env", { "modules": "commonjs" }],
        "react"
      ],
      "plugins": [
        "transform-object-rest-spread",
        "transform-class-properties",
        "transform-export-extensions",
        "transform-object-assign"
      ],
      "sourceMaps": "both"
    }
  }
}
```

not the worst either, but imo that's a bit less clear!

Reply:
Oh whoops, yeah I didn't realize. We can add a comment about why later.

Thanks for following up.